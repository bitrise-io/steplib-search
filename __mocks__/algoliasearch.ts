export default jest.fn((applicationID: string, apiKey: string) => ({
  initIndex: jest.fn()
}));
