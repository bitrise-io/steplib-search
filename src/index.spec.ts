import algoliasearch from 'algoliasearch';

import StepLib from '.';

describe(StepLib, () => {
  const appId = 'app-id',
    apiKey = 'api-key';

  beforeEach(() => {
    ((algoliasearch as any) as jest.Mock).mockClear();
  });

  it('creates a new object with default indices', () => {
    const steplib = new StepLib(appId, apiKey);
    expect(steplib).toBeDefined();
    expect(algoliasearch).toHaveBeenCalledWith(appId, apiKey);

    const [
      {
        value: { initIndex }
      }
    ] = ((algoliasearch as any) as jest.Mock).mock.results;
    expect(initIndex).toHaveBeenCalledWith('steplib_steps');
    expect(initIndex).toHaveBeenCalledWith('steplib_inputs');
  });

  it('can set custom idex values', () => {
    const stepIndex = 'some_step_index',
      inputsIndex = 'another_inputs_index';
    new StepLib(appId, apiKey, { stepIndex, inputsIndex });

    const [
      {
        value: { initIndex }
      }
    ] = ((algoliasearch as unknown) as jest.Mock).mock.results;
    expect(initIndex).toHaveBeenCalledWith(stepIndex);
    expect(initIndex).toHaveBeenCalledWith(inputsIndex);
  });
});
