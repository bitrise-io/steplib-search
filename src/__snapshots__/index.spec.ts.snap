// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepLib list list steps by step id 1`] = `
Array [
  Object {
    "cvs": "without-cvs@2.0.0",
    "latest": true,
  },
  Object {
    "cvs": "with-cvs@1.0.0",
    "latest": false,
  },
  Object {
    "cvs": "with-cvs@1.1.0",
    "latest": false,
  },
  Object {
    "cvs": "with-partial-cvs@2.1.0",
    "latest": true,
  },
]
`;

exports[`StepLib list list steps with inputs 1`] = `
Array [
  Object {
    "cvs": "activate-ssh-key@4.0.5",
    "inputs": Array [
      Object {
        "opts": Array [
          [Function],
        ],
        "ssh_rsa_private_key": "$SSH_RSA_PRIVATE_KEY",
      },
      Object {
        "opts": Array [
          [Function],
        ],
        "ssh_key_save_path": "$HOME/.ssh/bitrise_step_activate_ssh_key",
      },
    ],
    "objectID": "114723561",
  },
  Object {
    "cvs": "cocoapods-install@1.10.1",
    "inputs": Array [],
    "objectID": "114726881",
  },
  Object {
    "cvs": "create-android-emulator@1.1.6",
    "inputs": Array [],
    "objectID": "114727601",
  },
  Object {
    "cvs": "codecov@1.1.6",
    "inputs": Array [
      Object {
        "opts": Array [
          [Function],
        ],
        "ssh_key_save_path": "$HOME/.ssh/bitrise_step_activate_ssh_key",
      },
    ],
    "objectID": "114727221",
  },
]
`;
